inc/connector/namespace.php

```diff
@@ -11,6 +14,343 @@
 function bootstrap() {
 	add_filter( 'pre_update_option_cron', __NAMESPACE__ . '\\update_cron_array', 10, 2 );
 	add_filter( 'pre_option_cron', __NAMESPACE__ . '\\get_cron_array' );
+
+	// Filters introduced in WP 5.1.
+	add_filter( 'pre_schedule_event', __NAMESPACE__ . '\\pre_schedule_event', 10, 2 );
+	add_filter( 'pre_reschedule_event', __NAMESPACE__ . '\\pre_reschedule_event', 10, 2 );
+	add_filter( 'pre_unschedule_event', __NAMESPACE__ . '\\pre_unschedule_event', 10, 4 );
+	add_filter( 'pre_clear_scheduled_hook', __NAMESPACE__ . '\\pre_clear_scheduled_hook', 10, 3 );
+	add_filter( 'pre_unschedule_hook', __NAMESPACE__ . '\\pre_unschedule_hook', 10, 2 );
+	add_filter( 'pre_get_scheduled_event', __NAMESPACE__ . '\\pre_get_scheduled_event', 10, 4 );
+	add_filter( 'pre_get_ready_cron_jobs', __NAMESPACE__ . '\\pre_get_ready_cron_jobs' );
+}
+
+/**
+ * Schedule an event with Cavalcade.
+ *
+ * @param null|bool $pre   Value to return instead. Default null to continue adding the event.
+ * @param stdClass  $event {
+ *     An object containing an event's data.
+ *
+ *     @type string       $hook      Action hook to execute when the event is run.
+ *     @type int          $timestamp Unix timestamp (UTC) for when to next run the event.
+ *     @type string|false $schedule  How often the event should subsequently recur.
+ *     @type array        $args      Array containing each separate argument to pass to the hook's callback function.
+ *     @type int          $interval  The interval time in seconds for the schedule. Only present for recurring events.
+ * }
+ * @return null|bool True if event successfully scheduled. False for failure.
+ */
+function pre_schedule_event( $pre, $event ) {
```

Comment:
This will need to fire the `schedule_event` filter to match Core's existing behaviour allowing events to be changed prior to saving. (Work uses the hook to test cron events are registered properly so our tests started failing.)

Reply:
Added in 2c862020a9b7130940ce7a7c84198d44313e8d4c.

Per core, it allows the event to be changed after duplicates have been checked for.

Reply:
Aha, good catch @peterwilsoncc. I'll check for any other filters we're bypassing too